### PR TITLE
[System] Add RFC 5424 support to Auth datastream

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message

- Adds support for RFC 5424 to `system.auth` datastream using syslog filebeat processor.
   - For users directly utilising the assets such as ingest pipeline without agent/beat, changes are made to existing GROK patterns to support RFC 5424
- Add tags, improve `error.message`, and add `pipeline_error` to `event.kind` upon `on_failure`.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## How to test this PR locally

`elastic-package stack down && elastic-package build && elastic-package stack up --version=8.10.3 -d -v  && eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate -v`
```
--- Test results for package: system - START ---
╭─────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├─────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ system  │ auth        │ pipeline  │ test-5424.log            │ PASS   │  10.007542ms │
│ system  │ auth        │ pipeline  │ test-auth-rhel79.log     │ PASS   │   3.304583ms │
│ system  │ auth        │ pipeline  │ test-auth-ubuntu1204.log │ PASS   │  21.605917ms │
│ system  │ auth        │ pipeline  │ test-auth.log            │ PASS   │   4.547708ms │
│ system  │ auth        │ pipeline  │ test-multiline.log       │ PASS   │   3.199833ms │
│ system  │ auth        │ pipeline  │ test-secure-rhel7.log    │ PASS   │   3.846833ms │
│ system  │ auth        │ pipeline  │ test-timestamp.log       │ PASS   │   2.669458ms │
╰─────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: system - END   ---
Done
```
`elastic-package stack down && elastic-package build && elastic-package stack up --version=8.10.3 -d -v  && eval "$(elastic-package stack shellinit)" && elastic-package test system --generate -v`
```
--- Test results for package: system - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ system  │ auth        │ system    │ default   │ PASS   │ 37.581683125s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: system - END   ---
Done
```
## Related issues

- Relates https://github.com/elastic/sdh-beats/issues/3895

### CODE REVIEW

File: packages/system/changelog.yml [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+- version: "1.47.0"
+  changes:
+    - description: Add RFC 5424 support for Auth datastream
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/8103
 - version: "1.46.0"
   changes:
     - description: Added field `winlog.event_data.EnabledPrivilegeList` as type keyword to security data stream.
       type: enhancement
-      link: https://github.com/elastic/integrations/pull/8230
```
Comment (kcreddy) [inline packages/system/changelog.yml:6]: created by `elastic-package format`

File: packages/system/changelog.yml [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+  changes:
+    - description: Add RFC 5424 support for Auth datastream
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/8103
 - version: "1.46.0"
   changes:
     - description: Added field `winlog.event_data.EnabledPrivilegeList` as type keyword to security data stream.
       type: enhancement
-      link: https://github.com/elastic/integrations/pull/8230	
+      link: https://github.com/elastic/integrations/pull/8230
```
Comment (kcreddy) [inline packages/system/changelog.yml:11]: created by `elastic-package format`

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ processors:
       target_field: event.original
```
Comment (efd6) [inline packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml:7]: How do we ever end up with a `message` field? The stream config has a processor that does a rename from `message` to `event.original`.

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -7,6 +7,7 @@ processors:
...
       ignore_missing: true
   - grok:
+      if: ctx.log?.syslog == null
```
Comment (efd6) [inline packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml:10]: How is this ever null without a completely empty document given that the syslog processor doesn't have an `ignore_failure: true`?

**Review by efd6 — COMMENTED** [review]
I'm a little confused by some of this.

File: packages/system/_dev/deploy/docker/sample_logs/auth-sample.log [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+<85>1 2023-10-17T18:20:49.792553+02:00 test-server sudo 97641 - -  moe : TTY=pts/0 ; PWD=/home/moe ; USER=root ; COMMAND=/bin/sh -c echo BECOME-SUCCESS-tloxdklioldcmsfhtmtsxdbarqeuodij ; /usr/libexec/platform-python /home/moe/.ansible/tmp/ansible-tmp-1697559649.496464-9871-251889025803116/moe.py
+<86>1 2023-10-17T18:20:49.421098+02:00 test-server sshd 97522 - - pam_unix(sshd:session): session opened for user moe by (uid=0)
+<86>1 2023-10-17T18:20:49.345013+02:00 test-server systemd 97528 - - pam_unix(systemd-user:session): session opened for user moe by (uid=0)
+<86>1 2023-10-17T18:20:49.298055+02:00 test-server sshd 97522 - - Accepted publickey for moe from 10.128.176.58 port 43134 ssh2: ED25519 SHA256:B9urXskvGmmfbJmVQf0YyjCf2h2GfenEfd+yo9wcS9g
+<86>1 2023-10-17T18:19:03.481104+02:00 test-server systemd 92929 - - pam_unix(systemd-user:session): session closed for user nobody
+<86>1 2023-10-17T18:18:53.426094+02:00 test-server su 92923 - - pam_unix(su:session): session closed for user nobody
+<86>1 2023-10-17T18:18:51.766031+02:00 test-server su 92923 - - pam_unix(su:session): session opened for user nobody by (uid=0)
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/system/_dev/deploy/docker/sample_logs/auth-sample.log:15]: Reduced the repetition in the logs.

File: packages/system/data_stream/auth/_dev/test/pipeline/test-5424.log [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+<85>1 2023-10-17T18:20:49.792553+02:00 test-server sudo 97641 - -  moe : TTY=pts/0 ; PWD=/home/moe ; USER=root ; COMMAND=/bin/sh -c echo BECOME-SUCCESS-tloxdklioldcmsfhtmtsxdbarqeuodij ; /usr/libexec/platform-python /home/moe/.ansible/tmp/ansible-tmp-1697559649.496464-9871-251889025803116/moe.py
+<86>1 2023-10-17T18:20:49.421098+02:00 test-server sshd 97522 - - pam_unix(sshd:session): session opened for user moe by (uid=0)
+<86>1 2023-10-17T18:20:49.345013+02:00 test-server systemd 97528 - - pam_unix(systemd-user:session): session opened for user moe by (uid=0)
+<86>1 2023-10-17T18:20:49.298055+02:00 test-server sshd 97522 - - Accepted publickey for moe from 10.128.176.58 port 43134 ssh2: ED25519 SHA256:B9urXskvGmmfbJmVQf0YyjCf2h2GfenEfd+yo9wcS9g
+<86>1 2023-10-17T18:19:03.481104+02:00 test-server systemd 92929 - - pam_unix(systemd-user:session): session closed for user nobody
+<86>1 2023-10-17T18:18:53.426094+02:00 test-server su 92923 - - pam_unix(su:session): session closed for user nobody
+<86>1 2023-10-17T18:18:51.766031+02:00 test-server su 92923 - - pam_unix(su:session): session opened for user nobody by (uid=0)
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/system/data_stream/auth/_dev/test/pipeline/test-5424.log]: Reduced repetition

File: packages/system/data_stream/auth/agent/stream/log.yml.hbs [hunk 7]
```diff
@@ -25,6 +25,15 @@ publisher_pipeline.disable_host: true
...
 - add_locale: ~
+- rename:
+    fields:
+      - from: message
+        to: event.original
+    ignore_missing: true
+    fail_on_error: false
+- syslog:
+    field: event.original
+    ignore_missing: true
```
[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/system/data_stream/auth/agent/stream/log.yml.hbs:36]: Added `ignore_failure: true`

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml (hunk 3 above)
Comment (kcreddy) [reply packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml:8]: There are 2 different requests from 2 clients. One of them using `system.auth` datastream with agent. One of them would want to only use the ingest pipeline solely. So, they wouldn't be using `rename` and `syslog` processors in the input config. Need to accommodate for both of the processors here.

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (kcreddy) [reply packages/system/data_stream/auth/elasticsearch/ingest_pipeline/default.yml:11]: Thanks for noticing. I added `ignore_failure: true`

**Review by efd6 — APPROVED** [review]

File: packages/system/data_stream/auth/fields/fields.yml [hunk 8]
```diff
@@ -56,7 +56,35 @@
...
+      type: group
+      fields:
+        # To be moved to ecs.yml after ECS version upgrade
+        - name: version
+          type: keyword
 - description: "Operating system version as a raw string."
   ignore_above: 1024
   name: version
   type: keyword
+# To be moved to ecs.yml after ECS version upgrade
```
Comment (muthu-mps) [inline packages/system/data_stream/auth/fields/fields.yml:69]: I have created an [issue](https://github.com/elastic/integrations/issues/8296) to keep track of updating the ECS version and reference the additional `sys.log` fields from there.

**Review by muthu-mps — APPROVED** [review]

**Pull request merged**